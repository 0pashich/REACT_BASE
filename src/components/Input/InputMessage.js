import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import SendIcon from '@material-ui/icons/Send';
import Input from '@material-ui/core/Input';

//const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
    input: {
        padding: theme.spacing(0, 1),
        // margin: theme.spacing(2, 2),

    },
    form: {
        display: 'flex',

    },
}));


const InputMessage = (props) => {
    const { onSubmit } = props
    const classes = useStyles();
    //  const theme = useTheme();

    const [inputValue, setInputValue] = React.useState('')

    const handleChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if (onSubmit) {
            onSubmit(inputValue)
            setInputValue('')
        }
    }

    return (


        <form className={classes.form} onSubmit={handleSubmit}>
            <Input
                disableUnderline
                fullWidth
                required
                autoFocus
                multiline
                className={classes.input}
                variant="outlined"
                label="Сообщение"
                placeholder="Введите сообщение"
                value={inputValue}
                onChange={handleChange}
            />
            <IconButton edge="end" aria-label="send" type="submit" >
                <SendIcon />
            </IconButton>
        </form>

    )
}

export default InputMessage