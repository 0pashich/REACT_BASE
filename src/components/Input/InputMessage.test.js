import { render, screen } from "@testing-library/react";
import InputMessage from "./InputMessage";


describe("InputMessage", () => {


    it('matches snapshot online', () => {
        const component = render(<InputMessage author="Anton" text="Hello" />)

        expect(component).toMatchSnapshot()
    })

    // it('matches snapshot online', () => {
    //     const component = render(<Message author="Anton" text="Hello" />)

    //     expect(component).toMatchSnapshot()
    // })

    // it("should contain message text 'Hello'", () => {
    //     render(<Message author="Anton" text="Hello" />)

    //     const wrapper = screen.getByText(/Hello/i)

    //     expect(wrapper).toBeInTheDocument()
})
