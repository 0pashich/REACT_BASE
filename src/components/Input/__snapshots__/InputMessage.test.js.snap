// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMessage matches snapshot online 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiInputBase-root MuiInput-root makeStyles-input-1 MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-multiline MuiInput-multiline"
          label="Сообщение"
          required=""
          variant="outlined"
        >
          <textarea
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            placeholder="Введите сообщение"
            required=""
            rows="1"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
        </div>
        <button
          aria-label="send"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="makeStyles-form-2"
    >
      <div
        class="MuiInputBase-root MuiInput-root makeStyles-input-1 MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-multiline MuiInput-multiline"
        label="Сообщение"
        required=""
        variant="outlined"
      >
        <textarea
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          placeholder="Введите сообщение"
          required=""
          rows="1"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
          tabindex="-1"
        />
      </div>
      <button
        aria-label="send"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
